src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,92 @@
+        } else if (s.equals("bye")) {
+            storage.updateTasks(tasks.getTasks());
+            return Ui.goodbye();
+        } else {
+            try {
+                String[] splitStr = s.split(" ");
+
+                if (splitStr.length == 2 && (splitStr[0].equals("mark") || splitStr[0].equals("unmark"))) {
+                    try {
+                        int itemNumber = Integer.parseInt(splitStr[1]);
+                        if (splitStr[0].equals("mark")) {
+                            tasks.markDone(itemNumber - 1);
+                            return Ui.informTaskDone(tasks.getTask(itemNumber - 1));
+                        } else {
+                            tasks.markUndone(itemNumber - 1);
+                            return Ui.informTaskUndone(tasks.getTask(itemNumber - 1));
+                        }
+                    } catch (NumberFormatException e) {
+                        throw new WrongInputException();
+                    }
+                } else if (splitStr.length == 2 && splitStr[0].equals("delete")) {
+                    if (tasks.getSize() == 0) {
+                        return "No items already, what you want to delete?\n";
+                    } else {
+                        try {
+                            int itemNumber = Integer.parseInt(splitStr[1]);
+                            Task deletedTask = tasks.getTask(itemNumber - 1);
+                            tasks.deleteItem(itemNumber - 1);
+                            return Ui.informTaskDeleted(deletedTask, tasks.getSize());
+                        } catch (NumberFormatException e) {
+                            throw new WrongInputException();
+                        }
+                    }
+                } else if (splitStr[0].equals("todo") || splitStr[0].equals("deadline") || splitStr[0].equals("event")) {
+                    Task addedTask = new Task("");
+                    switch (splitStr[0]) {
+                    case "todo":
+                        String[] todoSplit = s.split("todo");
+                        if (todoSplit.length <= 1 || todoSplit[1].trim().equals("")) {
+                            throw new EmptyTodoException();
+                        }
+                        addedTask = new Todo(todoSplit[1].trim());
+                        break;
+                    case "deadline":
+                        String splitStr1 = s.split("deadline ")[1];
+                        String[] splitStr2 = splitStr1.split(" /by ");
+                        String[] splitStr3 = splitStr2[1].split(" ");
+                        LocalDateTime dateTime = null;
+                        if (splitStr3.length == 2) {
+                            dateTime = DateManager.parseDateString(splitStr3[0], splitStr3[1]);
+                        }
+                        addedTask = dateTime == null ? new Deadline(splitStr2[0], splitStr2[1])
+                                : new Deadline(splitStr2[0], dateTime);
```

Comment:
from the 2103 textbook! :

Here are two acceptable ways to format ternary expressions:
alpha = (aLongBooleanExpression) ? beta : gamma;
alpha = (aLongBooleanExpression)
        ? beta
        : gamma;